runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/io/SdkByteReadChannelJVM.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.io
+
+import kotlinx.coroutines.runBlocking
+import java.io.InputStream
+import java.util.*
+
+/**
+ * Create a blocking [InputStream] that blocks everytime the channel suspends at [SdkByteReadChannel.read]
+ */
+public fun SdkByteReadChannel.toInputStream(): InputStream = InputAdapter(this)
+
+private const val DEFAULT_READ_BYTES = 8192L
+private class InputAdapter(private val ch: SdkByteReadChannel) : InputStream() {
```

Comment:
Should we support `mark` / `reset`? I think it's not supported by default (`markSupported()` defaults to `false`). It can probably be additive work if requested

Reply:
I think I'd vote to make it additive based on customer feedback/requests. Anyone feel different?

Reply:
Sounds good to me.